### ISSUE TITLE
embark integration with consult-notes-org-headings-mode 

### ISSUE THREAD

**Issue by fhansen42** [mclear-tools/consult-notes#48]
consult-notes-org-headings-mode is great. It's allowed me to replace [consult-org-agenda](https://github.com/minad/consult/blob/main/consult-org.el) with consult-notes and get a combined interface for selecting org headings and file names. One items that is missing, however, is identification of the org heading as an [org-heading target](https://github.com/oantolin/embark/blob/d7957f624cf3ff20609ac4949115c8ba7d2e3d06/embark-org.el) within embark. Currently it is identified as a consult-note target after calling embark-act.

FYI to @oantolin and @minad in case either of them would like to offer any ideas on how to fix this.

**Comment by oantolin** [comment]
The text properrty with the marker should be called `org-marker` (rather than `consult--candidate` as it is currently) and the completion category should be `org-heading` (rather than `consult-notes-category` as it is currently). So it's a simple two-symbol change.

Also users can do this on their own without any changes to this library:

```elisp
(defun refine-consult-notes (_type target)
  (when-let ((marker (get-text-property 0 'consult--candidate target))
             ((markerp marker)))
    (cons 'org-heading (propertize target 'org-marker marker))))

(add-to-list 'embark-transformer-alist '(consult-notes . refine-consult-notes))
```

**Comment by oantolin** [comment]
> Currently it is identified as a consult-note target after calling embark-act.

Isn't it identified as a `consult-notes-category` target? (This is why I wouldn't end a category name in the word `category`.)

**Comment by minad** [comment]
@oantolin There seems to be a consult-notes-category defcustom with a value consult-note.

**Comment by oantolin** [comment]
Oh, I read very quickly and thought that symbol was quoted, @minad. Thanks for the correction.

**Comment by mclearc** [comment]
@fhansen42 is the solution offered by @oantolin enough? I could just put that in the readme.

**Comment by fhansen42** [comment]
I've added the code provided by @oantolin to my .init file. I don't see any change in the behavior of embark-act called on a heading in consult-notes. I've just appended to the bottom of my .init. I'm not sure if it makes a difference where it is placed?

**Comment by mclearc** [comment]
Yeah I'm not seeing a difference either. I'll look into this soon, when I have some more time.

**Comment by oantolin** [comment]
Oh, I'm sorry, that code won't work. `consult-notes` uses `consult--multi` internally, right? The embark support for `consult--multi` already uses `embark-transformer-alist` and Embark only does one round of transformation.

**Comment by oantolin** [comment]
It's funny but if you use `(setq consult-notes-category 'org-heading)`  then Embark actions work because of a coincidence: the marker is still in the text property `consult--candidate` so embark-org doesn't find it (it expects the marker to be stored in the text property `org-marker`), and thus embark-org doesn't jump to the heading before running the action, *but* thanks to consult-notes's preview *point is already at the heading* and the lack of a jump doesn't matter!

Now, if you collected some `consult-notes` candidates, then Embark actions wouldn't work from the embark collect buffer.

So I stand by my original recommendation: store the marker in `org-marker` rather than `consult--candidate` (or in both places if you need to) and set `consult-notes-category` to `org-heading` (maybe that should only be set in `consult-notes-org-heading-mode`; I'm not familiar with what this package does in any other mode).
